integration-tests/tests/test_identity_transformation.py

```diff
@@ -0,0 +1,122 @@
+    BaseConfig,
+    DatasetLogs,
+)
+from tests.utils.utils import (
+    diff_equal,
+    run_and_assert,
+)
+
+pytestmark = pytest.mark.binaries
+
+text_datasets = pytest.mark.parametrize(
+    "download_and_extract_dataset",
+    [
+        "hive_24hr",
+    ],
+    indirect=["download_and_extract_dataset"],
+)
+
+json_datasets = pytest.mark.parametrize(
+    "download_and_extract_dataset",
+    [
+        "spark_event_logs",
+        "postgresql",
+    ],
+    indirect=["download_and_extract_dataset"],
+)
+
+
+@pytest.mark.clp
+@text_datasets
+def test_clp_identity_transform(
+    request, base_config: BaseConfig, download_and_extract_dataset: DatasetLogs
+) -> None:
+    binary_path_str = str(base_config.clp_bin_dir / "clp")
+    dataset_name = download_and_extract_dataset.name
+    download_dir = base_config.uncompressed_logs_dir / dataset_name
+    archives_dir = base_config.test_output_dir / f"{dataset_name}-archives"
+    extract_dir = base_config.test_output_dir / f"{dataset_name}-logs"
+
+    shutil.rmtree(archives_dir, ignore_errors=True)
+    shutil.rmtree(extract_dir, ignore_errors=True)
+
+    # fmt: off
+    compression_cmd = [
+        binary_path_str,
+        "c",
+        "--progress",
+        "--remove-path-prefix", str(download_dir),
+        str(archives_dir),
+        str(download_dir),
+    ]
+    # fmt: on
+    run_and_assert(compression_cmd)
+    run_and_assert([binary_path_str, "x", str(archives_dir), str(extract_dir)])
+
+    diff_equal(download_dir, extract_dir)
+
+    shutil.rmtree(archives_dir, ignore_errors=True)
+    shutil.rmtree(extract_dir, ignore_errors=True)
+
+
+@pytest.mark.clp_s
+@json_datasets
+def test_clp_s_identity_transform(
+    request, base_config: BaseConfig, download_and_extract_dataset: DatasetLogs
+) -> None:
+    binary_path_str = str(base_config.clp_bin_dir / "clp-s")
+    dataset_name = download_and_extract_dataset.name
+    download_dir = base_config.uncompressed_logs_dir / dataset_name
+    archives_dir = base_config.test_output_dir / f"{dataset_name}-archives"
+    extract_dir = base_config.test_output_dir / f"{dataset_name}-logs"
+
+    shutil.rmtree(archives_dir, ignore_errors=True)
+    shutil.rmtree(extract_dir, ignore_errors=True)
+
+    run_and_assert([binary_path_str, "c", str(archives_dir), str(download_dir)])
+    run_and_assert([binary_path_str, "x", str(archives_dir), str(extract_dir)])
+
+    # Recompress the decompressed single-file output and decompress it again to verify consistency.
+    # TODO: Remove this check once we can directly compare decompressed logs (which would preserve
+    #       the directory structure and row/key order) with the original downloaded logs.
+    # See also: https://docs.yscope.com/clp/main/user-guide/core-clp-s.html#current-limitations
+    single_file_archives_dir = base_config.test_output_dir / f"{dataset_name}-single-file-archives"
+    single_file_extract_dir = base_config.test_output_dir / f"{dataset_name}-single-file-logs"
```

Comment:
```suggestion
    single_file_extract_dir = base_config.test_output_dir / f"{dataset_name}-single-file-extracted-logs"
```
It seems you don't have the single-file-archive flag on line 96's command though.

I feel we shouldn't mix the SFA and non-SFA between the first and 2nd time compression. can we make SFA to be a parameter. so we can run `test_clp_s_identity_transform` twice, one with SFA and one with Non-SFA?

Also, I feel we should make --single-file-archive to be

Reply:
@Bill-hbrhbr  any comment?

Reply:
See the latest change for my solution.